agent/command/s3_get.go

```diff
@@ -70,9 +74,18 @@ type s3get struct {
 	// for missing files.
 	Optional string `mapstructure:"optional" plugin:"expand"`
 
-	bucket      pail.Bucket
+	// TemporaryRoleARN is not meant to be used in production. It is used for testing purposes
+	// relating to the DEVPROD-5553 project.
+	// This is an ARN that should be assumed to make the S3 request.
+	// TODO (DEVPROD-13982): Upgrade this flag to RoleARN.
+	TemporaryRoleARN string `mapstructure:"temporary_role_arn" plugin:"expand"`
+
 	skipMissing bool
 
+	bucket          pail.Bucket
+	internalBuckets []string
+
+	taskdata client.TaskData
```

Comment:
On the other hand, is it necessary to store a `taskdata` field when it's only used in `Execute`? It would be easier to just make a local variable to store the task data and pass it into `comm.AssumeRole`, or just pass the struct inline to the function.

Reply:
That's fair, I was copying s3_put.go but it did end up being one call, so I put it back inline